### ISSUE TITLE
Error if parameter "type" exist in Update action

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/perseo-fe#30]
### LOG Message:
time=2015-01-28T14:11:22.897Z | lvl=ERROR | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=TypeError: Cannot read property '0' of undefined

### Rule created
```
------------------------------ Request ----------------------------------------------
url: (POST  http://10.95.169.246:9090/rules)

Header: {'Fiware-Service': 'unknownT', 'Content-Type': 'application/json', 'Fiware-ServicePath': u'/', 'Accept': 'application/json'}


Body: {"action": {"type": "update", "parameters": {"type": "message", "name": "ALARM", "value": "warning"}}, "text": "select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]", "name": "update_name"}


---------------------------------- Response ----------------------------------------------
status code: 200

Header: {'content-length': '1035', 'unica-correlator': '0f2a2db6-c776-4a5e-970d-c6f70c76d746', 'x-content-type-options': 'nosniff', 'x-powered-by': 'Express', 'connection': 'keep-alive', 'date': 'Wed, 28 Jan 2015 14:14:55 GMT', 'content-type': 'application/json; charset=utf-8'}

Body: {"error":null,"data":[false,[{"code":200,"body":{"text":"create context ctxt$unknownt$ partition by tenant from iotEvent(tenant=\"unknownt\" and service=\"/\")","timeLastStateChange":1422454282609,"name":"ctxt$unknownt$","state":"STARTED"}},{"code":200,"body":{"text":"context ctxt$unknownt$ select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","timeLastStateChange":1422454282609,"name":"update_name@unknownt/","state":"STARTED"}},null],{"action":{"type":"update","parameters":{"type":"message","name":"ALARM","value":"warning"}},"text":"select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","name":"update_name","service":"/","tenant":"unknownt","_id":"54c8eedf3a7608293a945da1"},null]}
```
### Notification
```
------------------------------ Request ----------------------------------------------
url: (POST  http://10.95.169.246:9090/notices)

Header: ({'Fiware-Service': 'unknownT', 'Content-Type': 'application/json', 'Fiware-ServicePath': u'/', 'Accept': 'application/json', 'User-Agent': 'orion/0.10.0'})

Body: {"originator": "localhost", "subscriptionId": "51c0ac9ed714fb3b37d7d5a8", "contextResponses": [{"contextElement": {"attributes": [{"type": "celcius", "name": "attrName_0", "value": "300", "metadatas": [{"type": "type_EHo0", "name": "name_LufR", "value": "jxdF"}]}], "type": "room", "id": "room2", "isPattern": "false"}, "statusCode": {"code": "200", "reasonPhrase": "OK"}}]})

---------------------------------- Response ----------------------------------------------
status code: 200

Header: {'content-length': '34', 'unica-correlator': '87de3f8d-5c59-4b79-8ca8-dc4212425b98', 'x-content-type-options': 'nosniff', 'x-powered-by': 'Express', 'connection': 'keep-alive', 'date': 'Wed, 28 Jan 2015 14:14:55 GMT', 'content-type': 'application/json; charset=utf-8'}

Body: ({"error":null,"data":{"code":200}})
```
### Mock
```
------------------------------ Request ----------------------------------------------
url: (GET  http://qa-orion-cep-01:9998/get/update)

---------------------------------- Response ----------------------------------------------
status code: 200

Header: {'date': 'Wed, 28 Jan 2015 14:15:01 GMT', 'content-length': '51', 'content-type': 'application/json', 'server': 'BaseHTTP/0.3 Python/2.6.6'}

Body: {"message": "Update context has not been sent yet"}
```

### Log commplete
```
time=2015-01-28T14:11:22.612Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=incoming request {"method":"POST","url":"/rules","headers":{"host":"10.95.169.246:9090","content-length":"362","accept-encoding":"gzip, deflate","fiware-service":"unknownT","fiware-servicepath":"/","accept":"application/json","user-agent":"python-requests/2.4.3 CPython/2.7.6 Windows/7","connection":"keep-alive","content-type":"application/json"},"body":{"action":{"type":"update","parameters":{"type":"message","name":"ALARM","value":"warning"}},"text":"select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","name":"update_name"}}
time=2015-01-28T14:11:22.620Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=making post to http://localhost:8080/perseo-core/rules
time=2015-01-28T14:11:22.635Z | lvl=DEBUG | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=post to http://localhost:8080/perseo-core/rules returns {"code":200,"body":{"text":"create context ctxt$unknownt$ partition by tenant from iotEvent(tenant=\"unknownt\" and service=\"/\")","timeLastStateChange":1422454282609,"name":"ctxt$unknownt$","state":"STARTED"}}
time=2015-01-28T14:11:22.636Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=done post to http://localhost:8080/perseo-core/rules
time=2015-01-28T14:11:22.636Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=making post to http://localhost:8080/perseo-core/rules
time=2015-01-28T14:11:22.649Z | lvl=DEBUG | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=post to http://localhost:8080/perseo-core/rules returns {"code":200,"body":{"text":"context ctxt$unknownt$ select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","timeLastStateChange":1422454282609,"name":"update_name@unknownt/","state":"STARTED"}}
time=2015-01-28T14:11:22.651Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=done post to http://localhost:8080/perseo-core/rules
time=2015-01-28T14:11:22.652Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=making post to http://demo-dca-be-01:8080/perseo-core/rules
time=2015-01-28T14:11:22.687Z | lvl=DEBUG | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=post to http://demo-dca-be-01:8080/perseo-core/rules returns {"code":200,"body":{"text":"create context ctxt$unknownt$ partition by tenant from iotEvent(tenant=\"unknownt\" and service=\"/\")","timeLastStateChange":1422454282649,"name":"ctxt$unknownt$","state":"STARTED"}}
time=2015-01-28T14:11:22.687Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=done post to http://demo-dca-be-01:8080/perseo-core/rules
time=2015-01-28T14:11:22.687Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=making post to http://demo-dca-be-01:8080/perseo-core/rules
time=2015-01-28T14:11:22.724Z | lvl=DEBUG | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=post to http://demo-dca-be-01:8080/perseo-core/rules returns {"code":200,"body":{"text":"context ctxt$unknownt$ select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","timeLastStateChange":1422454282649,"name":"update_name@unknownt/","state":"STARTED"}}
time=2015-01-28T14:11:22.725Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=done post to http://demo-dca-be-01:8080/perseo-core/rules
time=2015-01-28T14:11:22.731Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=rulesStore.Save "{\"action\":{\"type\":\"update\",\"parameters\":{\"type\":\"message\",\"name\":\"ALARM\",\"value\":\"warning\"}},\"text\":\"select *,\\\"update_name\\\" [...]"
time=2015-01-28T14:11:22.732Z | lvl=INFO | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=sending response: 200 {"error":null,"data":[false,[{"code":200,"body":{"text":"create context ctxt$unknownt$ partition by tenant from iotEvent(tenant=\"unknownt\" and service=\"/\")","timeLastStateChange":1422454282609,"name":"ctxt$unknownt$","state":"STARTED"}},{"code":200,"body":{"text":"context ctxt$unknownt$ select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","timeLastStateChange":1422454282609,"name":"update_name@unknownt/","state":"STARTED"}},null],{"action":{"type":"update","parameters":{"type":"message","name":"ALARM","value":"warning"}},"text":"select *,\"update_name\" as ruleName, \"/\" as service, \"unknownT\" as tenant, *, ev.attrName_0? as attrName_0_var, ev.id? as Meter from pattern [every ev=iotEvent(cast(cast(attrName_0?,String),float)>1.5 and type=\"room\")]","name":"update_name","service":"/","tenant":"unknownt","_id":"54c8ee0a3a7608293a945da0"},null]}
time=2015-01-28T14:11:22.735Z | lvl=DEBUG | corr=367b2409-cb6a-4665-9c04-25abe1fd1f5c | trans=367b2409-cb6a-4665-9c04-25abe1fd1f5c | op=/rules | msg=response-time: 123 statusCode: 200
time=2015-01-28T14:11:22.844Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=incoming request {"method":"POST","url":"/notices","headers":{"host":"10.95.169.246:9090","content-length":"374","accept-encoding":"gzip, deflate","fiware-service":"unknownT","fiware-servicepath":"/","accept":"application/json","user-agent":"orion/0.10.0","connection":"keep-alive","content-type":"application/json"},"body":{"originator":"localhost","subscriptionId":"51c0ac9ed714fb3b37d7d5a8","contextResponses":[{"contextElement":{"attributes":[{"type":"celcius","name":"attrName_0","value":"300","metadatas":[{"type":"type_Kwqq","name":"name_oEox","value":"ps5e"}]}],"type":"room","id":"room2","isPattern":"false"},"statusCode":{"code":"200","reasonPhrase":"OK"}}]}}
time=2015-01-28T14:11:22.845Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=making post to http://localhost:8080/perseo-core/events
time=2015-01-28T14:11:22.848Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=making post to http://demo-dca-be-01:8080/perseo-core/events
time=2015-01-28T14:11:22.865Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=incoming request {"method":"POST","url":"/actions/do","headers":{"content-type":"application/json; charset=utf-8","unica-correlator":"c62333c5-476d-4d57-8f30-e21148cbf714","cache-control":"no-cache","pragma":"no-cache","user-agent":"Java/1.7.0_67","host":"127.0.0.1:9090","accept":"text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2","connection":"keep-alive","content-length":"285"},"body":{"Meter":"room2","ev":{"attrName_0__type":"celcius","id":"room2","noticeId":"89e310d0-a6f7-11e4-a8ba-07c85e3a4cb6","tenant":"unknownt","service":"/","attrName_0":"300","type":"room","isPattern":"false"},"attrName_0_var":"300","tenant":"unknownT","ruleName":"update_name","service":"/"}}
time=2015-01-28T14:11:22.866Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=missing service header
time=2015-01-28T14:11:22.866Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=missing tenant header
time=2015-01-28T14:11:22.870Z | lvl=DEBUG | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=response-time: 5 statusCode: 200
time=2015-01-28T14:11:22.871Z | lvl=DEBUG | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=tenant unknownt, service /, event {"Meter":"room2","ev__attrName_0__type":"celcius","ev__id":"room2","ev__noticeId":"89e310d0-a6f7-11e4-a8ba-07c85e3a4cb6","ev__tenant":"unknownt","ev__service":"/","ev__attrName_0":"300","ev__type":"room","ev__isPattern":"false","attrName_0_var":"300","tenant":"unknownt","ruleName":"update_name","service":"/","id":"room2"}
time=2015-01-28T14:11:22.876Z | lvl=DEBUG | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=post to http://localhost:8080/perseo-core/events returns {"code":200}
time=2015-01-28T14:11:22.878Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=done post to http://localhost:8080/perseo-core/events
time=2015-01-28T14:11:22.878Z | lvl=INFO | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=sending response: 200 {"error":null,"data":{"code":200}}
time=2015-01-28T14:11:22.881Z | lvl=DEBUG | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=c62333c5-476d-4d57-8f30-e21148cbf714 | op=/notices | msg=response-time: 37 statusCode: 200
time=2015-01-28T14:11:22.897Z | lvl=ERROR | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=TypeError: Cannot read property '0' of undefined
time=2015-01-28T14:11:22.897Z | lvl=DEBUG | corr=c62333c5-476d-4d57-8f30-e21148cbf714 | trans=cf85f9d0-92aa-401a-b3e0-bfaa1621fc9c | op=/actions/do | msg=response-time: 32
time=2015-01-28T14:11:22.935Z | lvl=DEBUG | corr=n/a | trans=n/a | op=n/a | msg=post to http://demo-dca-be-01:8080/perseo-core/events returns {"code":200}
time=2015-01-28T14:11:22.935Z | lvl=INFO | corr=n/a | trans=n/a | op=n/a | msg=done post to http://demo-dca-be-01:8080/perseo-core/events

```

**Comment by fgalan** [comment]
Fixed in PR #31 (release/0.4.1) and #32 (develop).

**Comment by iariasleon** [comment]
LGTM
